src/vm/exceptionhandling.cpp

```diff
@@ -4585,6 +4594,29 @@ VOID DECLSPEC_NORETURN DispatchManagedException(PAL_SEHException& ex)
     }
 }
 
+VOID PALAPI HandleHardwareException(PAL_SEHException* ex)
+{
+    // Create frame necessary for the exception handling
+    FrameWithCookie<FaultingExceptionFrame> fef;
+#if defined(WIN64EXCEPTIONS)
+    *((&fef)->GetGSCookiePtr()) = GetProcessGSCookie();
+#endif // WIN64EXCEPTIONS
+    GCX_COOP();
+    fef.InitAndLink(&ex->ContextRecord);
+
+    // A hardware exception is handled only if it happened in a jitted code or 
+    // in one of the JIT helper functions (JIT_MemSet, ...)
+    PCODE controlPc = GetIP(&ex->ContextRecord);
+    if (ExecutionManager::IsManagedCode(controlPc) || IsIPInJitHelper(controlPc))
```

Comment:
I am not sure whether GC stackwalking is going to work correctly without AdjustContextForWriteBarrier. Not blocking for the change to be merged ... something to keep in mind for the GC stackwalking stress. Maybe add TODO about it?